api/api.proto

```diff
@@ -2684,3 +2686,74 @@ message LocateResponse {
     // <containerid>: /var/www
     map<string, string> dockerids = 2;
 }
+
+message VolumePlacementSpec {
+  // Defines a list of rules as part of the placement spec
+  repeated VolumePlacementRule rules = 1;
+}
+
+message VolumePlacementRule {
+  // Defines the number of volume replicas affected by this rule. If not provided,
+  // rule would affect all replicas
+  // +optional
+  uint64 affectedReplicas = 1;
+  // Defines the affinity rule for the volume placement
+  VolumePlacementAffinity affinity = 2;
+}
+
+// VolumePlacementAffinity allows to specify affinity and anti-affinity rules based on
+// storage pools, nodes and volumes.
+message VolumePlacementAffinity {
```

Comment:
Q: Can all these settings be set at the same time? If any of these settings cannot/should not be set together they should be in a `oneof`